src/reactive/consumer.py

```diff
@@ -44,15 +48,65 @@ def consume(mongodb_uri: str, queue_name: str) -> None:
     """
     with Connection(mongodb_uri) as conn:
         with closing(SimpleQueue(conn, queue_name)) as simple_queue:
-            msg = simple_queue.get(block=True)
-            try:
-                job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
-            except ValidationError as exc:
-                msg.reject(requeue=True)
-                raise JobError(f"Invalid job details: {msg.payload}") from exc
-            logger.info(
-                "Received job with labels %s and run_url %s",
-                job_details.labels,
-                job_details.run_url,
-            )
-            msg.ack()
+            with signal_handler(signal.SIGTERM):
+                msg = simple_queue.get(block=True)
+                try:
+                    job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
+                except ValidationError as exc:
+                    msg.reject(requeue=True)
+                    raise JobError(f"Invalid job details: {msg.payload}") from exc
+                logger.info(
+                    "Received job with labels %s and run_url %s",
+                    job_details.labels,
+                    job_details.run_url,
+                )
+                msg.ack()
```

Comment:
I think the message will only be rolled back on exit if it's not `ack`? Maybe this line, `msg.ack()`, needs to be placed outside of the `with` block?

Reply:
If we put it outside the block and we have an execution error in `msg.ack()`, this would mean that the message would not be recovered. So I think it should stay inside the block.